### PR TITLE
[deploy] Provide a flag to choose the container factory to use to run functions

### PR DESCRIPTION
**Pull request by sijie**
*Motivation*

Currently the default function running mode is `Process`. It is hardly able to configure running
functions in k8s.

Provide a flag in `gen-yml-from-env.py` so that people can choose which container factory to use.

### CODE REVIEW

File: docker/pulsar/scripts/gen-yml-from-env.py [hunk 1]
```diff
@@ -75,6 +75,19 @@
                 conf_to_modify = conf_to_modify[key_part]
                 modified = True
             i += 1
+
+    containerFactory=os.environ['PF_containerFactory']
```
Comment (cckellogg) [inline docker/pulsar/scripts/gen-yml-from-env.py]: What about this `os.environ.get('PF_containerFactory', None)`? The current way will throw an exception if the key does not exist and the script will fail and not update the file.

Comment (sijie) [reply docker/pulsar/scripts/gen-yml-from-env.py]: done

**Comment by sijie** [comment]
run java8 tests
run integration tests

**Comment by jerrypeng** [comment]
run java8 tests
run integration tests

**Comment by merlimat** [comment]
run java8 tests

**Pull request merged**